.github/actions/create-jira-ticket/action.yml

```
@@ -0,0 +1,140 @@
+name: Workflow incident tracking
+description: Create Jira ticket on incident
+
+inputs:
+  jira_base_url:
+    required: true
+    description: jira base url
+  jira_user_email:
+    required: true
+    description: jira user email
+  jira_api_token:
+    required: true
+    description: jira api token
+  module_name:
+    required: true
+    description: module name
+  reason_for_ticket_creation:
+    required: true
+    description: type of incident causing the ticket creation (currently veracode or nightly)
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+      if: ${{ inputs.reason_for_ticket_creation != 'veracode' }}
+
+    - name: Get ticket elements from context
+      id: get_context
+      run: |
+        # Change context elements (summary, parent epic, etc.) that is checked depending on the reason the ticket is created
+        if [[ "${{ inputs.reason_for_ticket_creation }}" == "veracode" ]]; then
+          parent_epic_id=83818
+          parent_epic_key="AT-268"
+          ticket_summary="PR-${{ github.event.pull_request.number }} incident on ${{ inputs.module_name }}"
+          ticket_labels="[\"Veracode\", \"Pipeline\"]"
+          run_url="N/A"
+        elif [[ "${{ inputs.reason_for_ticket_creation }}" == "nightly" ]]; then
+          parent_epic_id=206242
+          parent_epic_key="MON-151547"
+          ticket_summary="$(date '+%Y-%m-%d') ${{ inputs.module_name }}-${{ github.ref_name }} nightly build failure"
```

```suggestion
          ticket_summary="$(date '+%Y-%m-%d') ${{ inputs.module_name }}-${{ github.ref_name }} nightly build failure"
```
should not reflect user input